### ISSUE TITLE
az acr import  with --source and --registry is confusing.

### ISSUE THREAD

**Issue by ankurkhemani** [Azure/azure-cli#10243]
**Describe the bug**
az acr import

**To Reproduce**
`az acr import -h` has `--source` and `--registry` parameters. The help section doesn't document that `-r` and `--source` with registry is mutually exclusive (i.e. both can't specify registry/loginserver name). This causes confusion and bad user experience for cx, running into error state not knowing what's the issue. The error message should also be refined to include the issue above when user actually runs into that condition.

Our [official doc](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-import-images#import-from-a-registry-in-a-different-subscription
) does say: `Notice that the --source parameter specifies only the source repository and image name, not the registry login server name.`

**Expected behavior**
Proper help and error message.

**Environment summary**
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli 2.0.70 *
```

**Comment by ankurkhemani** [comment]
// cc @hkuadithya

**Comment by yugangw-msft** [comment]
Also consider to handling the case if --source uses the same registry, that we should error out. This happened once on our Cx

**Comment by adewaleo** [comment]
cc: @Wwwsylvia @yugangw-msft

**Comment by yugangw-msft** [comment]
This will be fixed by #12375
